sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -860,10 +860,11 @@ ssize_t coap_opt_add_uquery2(coap_pkt_t *pkt, const char *key, size_t key_len,
     assert(!val_len || (val && val_len));
 
     char qs[NANOCOAP_QS_MAX];
+    /* length including '=' */
     size_t qs_len = key_len + ((val && val_len) ? (val_len + 1) : 0);
 
     /* test if the query string fits */
-    if (qs_len >= NANOCOAP_QS_MAX) {
+    if (qs_len > NANOCOAP_QS_MAX) {
```

Comment:
This should be `>=` according to https://github.com/RIOT-OS/RIOT/pull/13022#discussion_r386328610 or is it different when using `_add_opt_pkt`?

Reply:
We no longer need to include the null terminator in the query string, so `qs_len` is exactly the length required for `qs`. So it's OK if `qs` is exactly the size of the buffer.